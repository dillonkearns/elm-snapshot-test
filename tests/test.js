const path = require('path');
const fs = require('fs');
const runElm = require('../run-elm');

// const getFiles = (/** @type {string} */ source) =>
//     fs.readdirSync(source, { withFileTypes: true })
//         .filter(dirent => dirent.isFile())
//         .map(dirent => dirent.name)
// 
// getFiles(path.join(__dirname, 'json-files')).forEach(jsonFilename => {
//     const json = JSON.parse(fs.readFileSync(path.join(__dirname, 'json-files', jsonFilename)))
//     Object.keys(json).forEach(key => {
//         test(`${key}`, async () => {
//             expect(json[key]).toMatchSnapshot();
//         });
//     })
// })
test('elm-snapshot-test', async () => {
    let snapshotValue = await runElm('src/Main.elm')
    expect(snapshotValue.value).toMatchSnapshot();
});