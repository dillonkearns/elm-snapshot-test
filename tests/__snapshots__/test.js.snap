// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elm-snapshot-test 1`] = `
Object {
  "\\"Aged Brie\\", 0, 0": "[Item \\"Aged Brie\\" -1 2]",
  "\\"Aged Brie\\", 0, 1": "[Item \\"Aged Brie\\" -1 3]",
  "\\"Backstage passes to a TAFKAL80ETC concert\\", 0, 0": "[Item \\"Backstage passes to a TAFKAL80ETC concert\\" -1 0]",
  "\\"Backstage passes to a TAFKAL80ETC concert\\", 0, 1": "[Item \\"Backstage passes to a TAFKAL80ETC concert\\" -1 0]",
  "\\"Not a special item\\", 0, 0": "[Item \\"Not a special item\\" -1 0]",
  "\\"Not a special item\\", 0, 1": "[Item \\"Not a special item\\" -1 0]",
  "\\"Sulfuras, Hand of Ragnaros\\", 0, 0": "[Item \\"Sulfuras, Hand of Ragnaros\\" 0 0]",
  "\\"Sulfuras, Hand of Ragnaros\\", 0, 1": "[Item \\"Sulfuras, Hand of Ragnaros\\" 0 1]",
}
`;
